demos/LowAllocationWebServer/ByteSpanExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+
+namespace LowAllocationServer
+{
+    public static class ByteSpanExtensions
+    {
+        internal static ByteSpan SliceTo(this ByteSpan buffer, char terminator, out int consumedBytes)
+        {
+            return buffer.SliceTo((byte) terminator, out consumedBytes);
```

Comment:
@krwq, is adding support to check the char. I will add a TODO and add it once his changes are done.